test/support/fixtures.ex

```diff
@@ -185,4 +186,27 @@ defmodule Beacon.Fixtures do
     |> PageEventHandler.changeset(full_attrs)
     |> Repo.insert!()
   end
+
+  def error_page_fixture(attrs \\ %{})
+
+  def error_page_fixture(%{layout: %Content.Layout{}} = attrs) do
+    {layout, attrs} = Map.pop!(attrs, :layout)
+
+    attrs
+    |> Map.put(:layout_id, layout.id)
+    |> error_page_fixture()
+  end
+
+  def error_page_fixture(attrs) do
+    full_attrs = %{
+      site: attrs[:site] || "my_site",
+      status: attrs[:status] || Enum.random(111..999),
+      template: attrs[:template] || "Uh-oh!",
+      layout_id: attrs[:layout_id] || layout_fixture().id
+    }
+
+    %ErrorPage{}
+    |> ErrorPage.changeset(full_attrs)
+    |> Repo.insert!()
+  end
```

Comment:
you could rewrite as:

```suggestion
  def error_page_fixture(attrs \\ %{}) do
    layout_id = get_lazy(attrs, :layout_id, fn -> layout_fixture().id end)

    attrs
    |> Enum.into(%{
      site: "my_site",
      status: Enum.random(400..599),
      template: "Uh-oh!",
      layout_id: layout_id
    })
    |> Content.create_error_page!()
  end
```

this way you can call the fixture as `error_page_fixture(status: 404)` or with a map if you wish while keeping default values with less lines of code :)

Reply:
The purpose of the extra clause here is to handle `error_page_fixture(%{layout: layout})`.  Almost all of the tests will have the full layout available, so we probably want to allow the fixture to be called that way.

Reply:
Passing `layout` sounds good, in that case it can be `layout_id = get_lazy(attrs, :layout, fn -> layout_fixture() end).id` that will have the same result